### PR TITLE
avoid deprecated methods since sbt 0.13.13

### PR DESCRIPTION
**Pull request by xuwei-k**
https://github.com/sbt/sbt/pull/2716

/cc @eed3si9n

### CODE REVIEW

**Comment by wsargent** [comment]
It looks like the documentation server breaks with this PR:

```
cd documentation
sbt run
```

Prior to this commit, this would bring up a documentation server on http://localhost:9000, but after this it goes to command prompt.

File: framework/project/Docs.scala [hunk 1]
```diff
@@ -26,28 +26,30 @@ object Docs {
...
       val docMappings = filtered.get pair rebase(docBase, "play/docs/content/")
 
       val apiDocMappings = (apiBase ** "*").get pair rebase(apiBase, "play/docs/content/api")
 
       // The play version is added so that resource paths are versioned
-      val webjarMappings = webjars.*** pair rebase(webjars, "play/docs/content/webjars/" + playVersion)
+      val webjarMappings = webjars.*** pair rebase(webjars, "play/docs/content/webjars/" + version.value)
 
       // Gather all the conf files into the project
-      val referenceConfMappings = confs.map {
```
Comment (wsargent) [inline framework/project/Docs.scala]: confs vs allConfs?

Comment (xuwei-k) [inline framework/project/Docs.scala]: Oops, I’m sorry I made a mistake

Comment (dwijnand) [inline framework/project/Docs.scala]: allConfs is correct, the key is allConfs and it's value was being bound as "confs"

File: framework/project/Build.scala [hunk 2]
```diff
@@ -136,7 +136,7 @@ object BuildSettings {
   )
 
   def playScriptedSettings: Seq[Setting[_]] = Seq(
-    ScriptedPlugin.scripted <<= ScriptedPlugin.scripted.tag(Tags.Test),
+    ScriptedPlugin.scripted := ScriptedPlugin.scripted.tag(Tags.Test).value,
```
Comment (wsargent) [inline framework/project/Build.scala]: use `.evaluated`

File: framework/project/Docs.scala [hunk 3]
```diff
@@ -26,28 +26,30 @@ object Docs {
   lazy val settings = Seq(
     apiDocsInclude := false,
     apiDocsIncludeManaged := false,
-    apiDocsScalaSources <<= (thisProjectRef, buildStructure) flatMap allSources(Compile, ".scala"),
-    apiDocsClasspath <<= (thisProjectRef, buildStructure) flatMap allClasspaths,
-    apiDocsJavaSources <<= (thisProjectRef, buildStructure) flatMap allSources(Compile, ".java"),
+    apiDocsScalaSources := ((thisProjectRef, buildStructure) flatMap allSources(Compile, ".scala")).value,
```
Comment (wsargent) [inline framework/project/Docs.scala]: use `.evaluated` here

File: framework/project/Docs.scala [hunk 4]
```diff
@@ -26,28 +26,30 @@ object Docs {
...
+    apiDocsClasspath := ((thisProjectRef, buildStructure) flatMap allClasspaths).value,
```
Comment (wsargent) [inline framework/project/Docs.scala]: use `.evaluated`

File: framework/project/Docs.scala [hunk 5]
```diff
@@ -26,28 +26,30 @@ object Docs {
...
+    apiDocsJavaSources := ((thisProjectRef, buildStructure) flatMap allSources(Compile, ".java")).value,
```
Comment (wsargent) [inline framework/project/Docs.scala]: use `.evaluated`

File: framework/project/Docs.scala [hunk 6]
```diff
@@ -26,28 +26,30 @@ object Docs {
...
     apiDocsUseCache := true,
-    apiDocs <<= apiDocsTask,
+    apiDocs := apiDocsTask.value,
```
Comment (wsargent) [inline framework/project/Docs.scala]: use `.evaluated`

**Comment by wsargent** [comment]
Hi @xuwei-k it looks like this might be from using `.value` rather than `.evaluated` -- `value` will return a task value without actually evaluating the transitive input task, while .evaluated will evaluate the transitive input task given the input.

File: framework/src/play-docs-sbt-plugin/src/main/scala/com/typesafe/play/docs/sbtplugin/PlayDocsPlugin.scala [hunk 7]
```diff
@@ -80,28 +80,28 @@ object PlayDocsPlugin extends AutoPlugin {
   def docsRunSettings = Seq(
     playDocsValidationConfig := ValidationConfig(),
     manualPath := baseDirectory.value,
-    run <<= docsRunSetting,
-    generateMarkdownRefReport <<= PlayDocsValidation.generateMarkdownRefReportTask,
-    validateDocs <<= PlayDocsValidation.validateDocsTask,
-    validateExternalLinks <<= PlayDocsValidation.validateExternalLinksTask,
+    run := docsRunSetting.value,
```
Comment (eed3si9n) [inline framework/src/play-docs-sbt-plugin/src/main/scala/com/typesafe/play/docs/sbtplugin/PlayDocsPlugin.scala]: This is the culprit. Sorry I didn't catch this during the review.

File: framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala [hunk 8]
```diff
@@ -48,21 +48,21 @@ object PlayForkRun extends AutoPlugin {
     libraryDependencies += "com.typesafe.play" %% "fork-run" % play.core.PlayVersion.current % ForkRun.name,
     PlaySettings.manageClasspath(ForkRun),
 
-    playRun <<= PlayRun.playDefaultRunTask,
-    playForkOptions <<= forkOptionsTask,
-    playForkRun <<= forkRunTask,
+    playRun := PlayRun.playDefaultRunTask.value,
```
Comment (eed3si9n) [inline framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala]: Use `.evaluated` here.

File: framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala [hunk 9]
```diff
@@ -48,21 +48,21 @@ object PlayForkRun extends AutoPlugin {
...
+    playForkOptions := forkOptionsTask.value,
+    playForkRun := forkRunTask.value,
```
Comment (eed3si9n) [inline framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala]: Use .evaluated here.

File: framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala [hunk 10]
```diff
@@ -48,21 +48,21 @@ object PlayForkRun extends AutoPlugin {
...
 
-    run in Compile <<= selectRunTask,
-    BackgroundJobServiceKeys.backgroundRun in Compile <<= backgroundForkRunTask,
+    run in Compile := selectRunTask.value,
```
Comment (eed3si9n) [inline framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala]: Use .evaluated here.

File: framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala [hunk 11]
```diff
@@ -48,21 +48,21 @@ object PlayForkRun extends AutoPlugin {
...
+    BackgroundJobServiceKeys.backgroundRun in Compile := backgroundForkRunTask.evaluated,
```
Comment (eed3si9n) [inline framework/src/sbt-fork-run-plugin/src/main/scala/play/sbt/forkrun/PlayForkRun.scala]: Use .evaluated here.

File: framework/src/sbt-plugin/src/main/scala/play/sbt/PlaySettings.scala [hunk 12]
```diff
@@ -90,7 +89,7 @@ object PlaySettings {
     },
 
     // THE `in Compile` IS IMPORTANT!
-    Keys.run in Compile <<= PlayRun.playDefaultRunTask,
+    Keys.run in Compile := PlayRun.playDefaultRunTask.value,
```
Comment (eed3si9n) [inline framework/src/sbt-plugin/src/main/scala/play/sbt/PlaySettings.scala]: Here too.

File: framework/src/play-docs-sbt-plugin/src/main/scala/com/typesafe/play/docs/sbtplugin/PlayDocsPlugin.scala (hunk 7 above)
Comment (eed3si9n) [inline framework/src/play-docs-sbt-plugin/src/main/scala/com/typesafe/play/docs/sbtplugin/PlayDocsPlugin.scala]: Use .evaluated here.

File: framework/project/Build.scala [hunk 13]
```diff
@@ -228,7 +228,7 @@ object PlayBuild extends Build {
     .settings(
       libraryDependencies ++= runtime(scalaVersion.value) ++ scalacheckDependencies,
 
-      sourceGenerators in Compile <+= (version, scalaVersion, sbtVersion, sourceManaged in Compile) map PlayVersion,
+      sourceGenerators in Compile += Def.task(((version, scalaVersion, sbtVersion, sourceManaged in Compile) map PlayVersion).value).taskValue,
```
Comment (dwijnand) [inline framework/project/Build.scala]: @xuwei-k no need to wrap and unwrap here (and similar places in this diff), this should work:

```scala
sourceGenerators in Compile += ((version, scalaVersion, sbtVersion, sourceManaged in Compile) map PlayVersion).taskValue
```

**Pull request merged**